File: src/tarea8/Libreria.java
package tarea8;
import java.util.ArrayList;
public class Libreria {
    private ArrayList<Books>libros=new ArrayList<Books>();
    private ArrayList<Usuarios>usuarios=new ArrayList<Usuarios>();
    private ArrayList<Books>librosRentados=new ArrayList<Books>();
    private ArrayList<Books>librosDisponibles=new ArrayList<Books>();
    private ArrayList<Usuarios>usuariosActivos=new ArrayList<Usuarios>();

    Libreria() {
        
    }
    
    public void mostrarLibros(){
        for (Books librosInterpreter : libros){
            System.out.println(librosInterpreter.mostrarLibros());
        }             
    }
    public void mostrarUsuarios(){
        for (Usuarios usuariosInterpreter : usuarios){
            System.out.println(usuariosInterpreter.mostrarUsuarios());
        }   
    }
    public void registrarUsuario(Usuarios newUsuario) {
        usuarios.add(newUsuario);
    }
    public void registrarLibro(Books newBook) {
        libros.add(newBook);
    }   
    public int rentarLibro (Books tituloRentadoID) {
        librosRentados.add(tituloRentadoID);
        return 0;
    } 

    public ArrayList<Books> getLibrosDisponibles() {
        return librosDisponibles;
    }    
    public static Books buscarLibroPorCodigo(ArrayList<Books> librosDisponibles, int buscarID) {
        for (Books libro : librosDisponibles) {
            if (libro.getLibroID() == buscarID) {
                return libro; 
            }
        }
        return null;
    }        
}
           

File: Libreria.py
class Libreria:
    def __init__(self):
        self.libros = []
        self.usuarios = []
        self.libros_rentados = []
        self.libros_disponibles = []
        self.usuarios_activos = []

    def registrar_usuario(self, new_usuario: Usuarios):
        self.usuarios.append(new_usuario)

    def registrar_libro(self, new_book: Libros):
        self.libros.append(new_book)

    def rentar_libro(self, titulo_rentado_id: int):
        for libro in self.libros:
            if libro.get_libro_id() == titulo_rentado_id and libro.is_libro_disponible():
                self.libros_rentados.append(libro)
                libro.set_libro_disponible(False)
                return 0                
        return none

    def get_libros_disponibles(self):
        return self.libros_disponibles

    def buscar_libro_por_codigo(libros_disponibles: List[Libros], buscar_id: int):
        for libro in libros_disponibles:
            if libro.get_libro_id() == buscar_id:
                return libro
        return None